src/commands/install.yml

```diff
@@ -21,94 +16,129 @@ steps:
 
-              export PIP=$(which pip pip3 | head -1)
-              if [[ -n $PIP ]]; then
-                if which sudo > /dev/null; then
-                  sudo $PIP install awscli --upgrade
-                else
-                  # This installs the AWS CLI to ~/.local/bin. Make sure that ~/.local/bin is in your $PATH.
-                  $PIP install awscli --upgrade --user
-                fi
-              elif [[ $(which unzip curl | wc -l) -eq 2 ]]; then
-                cd
-                curl "https://s3.amazonaws.com/aws-cli/awscli-bundle.zip" -o "awscli-bundle.zip"
-                unzip awscli-bundle.zip
-                if which sudo > /dev/null; then
-                  sudo ~/awscli-bundle/install -i /usr/local/aws -b /usr/local/bin/aws
-                else
-                  # This installs the AWS CLI to the default location (~/.local/lib/aws) and create a symbolic link (symlink) at ~/bin/aws. Make sure that ~/bin is in your $PATH.
-                  awscli-bundle/install -b ~/bin/aws
-                fi
-                rm -rf awscli-bundle*
-                cd -
-              else
-                echo "Unable to install AWS CLI. Please install pip."
-                exit 1
-              fi
-          ;;
-          "2")
-            if [ "<<parameters.skip-install-check>>" == "false" ] && which aws > /dev/null; then
-                echo "The AWS CLI is already installed. Skipping."
-                exit 0
-              fi
+        if [[ $EUID == 0 ]]; then export SUDO=""; else export SUDO="sudo"; fi
 
-              if [[ $EUID == 0 ]]; then export SUDO=""; else export SUDO="sudo"; fi
+        if ! command -v aws --version >/dev/null 2>&1  ; then
+            echo AWS is not installed
+        else
+            echo AWS is currently installed.
+            if aws --version 2>&1 | grep -q $AWS_VER_REGEXP_2; then
+                echo AWS CLI v2 is installed
+                AWS_CLI_INSTALLED_VERSION="2"
+            fi
+            if aws --version 2>&1 | grep -q $AWS_VER_REGEXP_1; then
+                echo AWS CLI v1 is installed
+                AWS_CLI_INSTALLED_VERSION="1"
+            fi
+        fi
 
-              cd /tmp || exit
+        #If the desired version of the CLI is not installed, install it.
+        if [[ $AWS_CLI_VERSION_SELECTED != $AWS_CLI_INSTALLED_VERSION ]]; then
 
-              # PLATFORM CHECK: mac vs. alpine vs. other linux
-              if uname -a | grep Darwin; then
-                SYS_ENV_PLATFORM=darwin
-              elif uname -a | grep Linux; then
-                SYS_ENV_PLATFORM=linux
-              else
-                echo "This platform appears to be unsupported."
-                uname -a
-                exit 1
-              fi
+            #uninstall AWS CLI if it is installed.
+            if which aws; then
+                echo Uninstalling old CLI
+                $SUDO rm -rf /usr/local/aws
+                $SUDO rm /usr/local/bin/aws
```

Comment:
Should we use `rm -f` in case the file doesn't exist?

Reply:
I have changed the logic to specifically remove the `$(which aws)` path. Hows that sound?

New test for this: https://app.circleci.com/pipelines/github/CircleCI-Public/aws-cli-orb/248/workflows/0de1a3ba-575d-4b9d-af93-a6321277ac5a/jobs/853